### PR TITLE
StratisX integration test capability

### PR DESCRIPTION
**Pull request by zeptin**
We should probably have at least a few regression tests to ensure that SBFN remains compatible with stratisX for the forseeable future. This PR is a first step towards that.

Note that this PR does not include binaries for Linux or OSX yet (I can only realistically provide the Linux one myself). Ideally they should be rebuilt from source in a controlled environment and possibly signed, as unknown binaries are a security risk.

The stratisd.exe included here has a necessary fix by Maciej to allow operation on regtest. This would need to be merged into stratisX master.

### CODE REVIEW

**Comment by bokobza** [comment]
Excellent stuff @zeptin !
Maybe @dev0tion is the man to ask for Linux and Mac runnables.

**Comment by bokobza** [comment]
> they should be rebuilt from source in a controlled environment and possibly signed, as unknown binaries are a security risk.

if we put them there and no one touches them on GitHub, it'll be fairly safe.

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs [hunk 1]
```diff
@@ -212,6 +214,26 @@ private void StartBitcoinCoreRunner()
                 failureReason: $"Failed to invoke GetBlockHash on BitcoinCore instance after {duration}");
         }
 
+        private void StartStratisXRunner()
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs]: interesting, these 2 methods `StartBitcoinCoreRunner` and `StartStratisXRunner` don't actually start the nodes, but rather wait for them to be up. the starting is done is `Start()`.
Can we merge them into 1 method (they're really the same) and call it somethingmore appropriate?
Then in `Start()` we'll have `if (this.runner is BitcoinCoreRunner || this.runner is StratisXRunner) WaitUntilRunning()` (please find a better name lol).

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs [hunk 2]
```diff
@@ -77,6 +77,23 @@ private static string GetBitcoinCorePath(string version)
             throw new FileNotFoundException($"Could not load the file {path}.");
         }
 
+        private static string GetStratisXPath(string version)
+        {
+            string path;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                path = $"../../../../External libs/StratisX/{version}/Windows/stratisd.exe";
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs]: That's definitely gonna fail for Linux and Mac so in the meantime, can you surround the inside of your test with ` if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {`

**Comment by zeptin** [comment]
Thanks @bokobza, I will make the requested changes.

There does seem to be a minor bug with the location of the `debug.log` file, I need to check the X source to see if this is fixable or if the debug log should just be entirely disabled (it's useful for troubleshooting tests though).

File: src/Stratis.Bitcoin.IntegrationTests/Compatibility/StratisXTests.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        }
+
+        /// <summary>
+        /// Tests whether a transaction relayed by SBFN appears in the
+        /// stratisX mempool. The transaction is then mined into a
+        /// block by SBFN, and the block must be accepted by stratisX.
+        /// </summary>
+        /// <remarks>Takes a while to run as block generation cannot
+        /// be sped up for this test.</remarks>
+        [Fact]
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/Compatibility/StratisXTests.cs]: Can you please mark this as unstable or skip it?

**Comment by dev0tion** [comment]
Hey @zeptin, whenever you feel like you're ready I'm always happy to provide an OS X build.
Linux you'll need to build yourself, but it's fairly easy to do so.
Hit me up if you need any help.

**Pull request merged**